File: solutions/csharp/allergies/1/Allergies.cs
[Flags]
public enum Allergen
{
    Eggs = 1,
    Peanuts = 2,
    Shellfish = 4,
    Strawberries = 8,
    Tomatoes = 16,
    Chocolate = 32,
    Pollen = 64,
    Cats = 128
}

public class Allergies
{
    private Allergen _allergies;
    public Allergies(int mask)
    {
        _allergies = (Allergen)mask;
    }

    public bool IsAllergicTo(Allergen allergen)
    {
        return _allergies.HasFlag(allergen);
    }

    public Allergen[] List()
    {
        return Enum.GetValues<Allergen>().Where(x => _allergies.HasFlag(x)).Select(x => (Allergen)x).ToArray();
    }
}

File: solutions/python/allergies/1/allergies.py
ALLERGENS = {
    "eggs": 1,
    "peanuts": 2,
    "shellfish": 4,
    "strawberries": 8,
    "tomatoes": 16,
    "chocolate": 32,
    "pollen": 64,
    "cats": 128
}

class Allergies:
    def __init__(self, score):
        self.score = score

    def allergic_to(self, item):
        return bool(self.score & ALLERGENS[item.lower()])

    @property
    def lst(self):
        return [x for x in ALLERGENS if self.allergic_to(x)]
